backend/src/main/java/ai/verta/modeldb/artifactStore/storageservice/s3/S3Client.java

```diff
@@ -0,0 +1,219 @@
+      initializetWithTemporaryCredentials(awsRegion);
+    } else {
+      LOGGER.debug("environment credentials based s3 client");
+      // reads credential from OS Environment
+      initializetWithEnvironment(awsRegion);
+    }
+  }
+
+  private void initializetWithEnvironment(Regions awsRegion) {
+    this.s3Client = AmazonS3ClientBuilder.standard().withRegion(awsRegion).build();
+  }
+
+  private void initializeMinioClient(
+      String cloudAccessKey, String cloudSecretKey, Regions awsRegion, String minioEndpoint) {
+    AWSCredentials awsCreds = new BasicAWSCredentials(cloudAccessKey, cloudSecretKey);
+    ClientConfiguration clientConfiguration = new ClientConfiguration();
+    clientConfiguration.setSignerOverride("AWSS3V4SignerType");
+
+    this.s3Client =
+        AmazonS3ClientBuilder.standard()
+            .withEndpointConfiguration(
+                new AwsClientBuilder.EndpointConfiguration(minioEndpoint, awsRegion.getName()))
+            .withPathStyleAccessEnabled(true)
+            .withClientConfiguration(clientConfiguration)
+            .withCredentials(new AWSStaticCredentialsProvider(awsCreds))
+            .build();
+  }
+
+  private void initializeS3ClientWithAccessKey(
+      String cloudAccessKey, String cloudSecretKey, Regions awsRegion) {
+    BasicAWSCredentials awsCreds = new BasicAWSCredentials(cloudAccessKey, cloudSecretKey);
+    this.s3Client =
+        AmazonS3ClientBuilder.standard()
+            .withRegion(awsRegion)
+            .withCredentials(new AWSStaticCredentialsProvider(awsCreds))
+            .build();
+  }
+
+  private void initializetWithTemporaryCredentials(Regions awsRegion) throws IOException {
+    String roleSessionName = "modelDB" + UUID.randomUUID().toString();
+
+    AWSSecurityTokenService stsClient = null;
+    AmazonS3 newS3Client = null;
+    try {
+      stsClient = AWSSecurityTokenServiceClientBuilder.standard().withRegion(awsRegion).build();
+
+      String roleArn = System.getenv(ModelDBConstants.AWS_ROLE_ARN);
+
+      String token =
+          new String(
+              Files.readAllBytes(
+                  Paths.get(
+                      ModelDBUtils.appendOptionalTelepresencePath(
+                          System.getenv(ModelDBConstants.AWS_WEB_IDENTITY_TOKEN_FILE)))));
+
+      // Obtain credentials for the IAM role. Note that you cannot assume the role of
+      // an AWS root account;
+      // Amazon S3 will deny access. You must use credentials for an IAM user or an
+      // IAM role.
+      AssumeRoleWithWebIdentityRequest roleRequest =
+          new AssumeRoleWithWebIdentityRequest()
+              .withDurationSeconds(900)
+              .withRoleArn(roleArn)
+              .withWebIdentityToken(token)
+              .withRoleSessionName(roleSessionName);
+
+      LOGGER.debug("assuming role");
+      // Call STS to assume the role
+      AssumeRoleWithWebIdentityResult roleResponse =
+          stsClient.assumeRoleWithWebIdentity(roleRequest);
```

Comment:
If the `token` is invalid at the starting time then the backend throw error like 
```
org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 's3Controller': Unsatisfied dependency expressed through field 's3Service'; nested exception is org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'getS3Service' defined in ai.verta.modeldb.App: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [ai.verta.modeldb.artifactStore.storageservice.s3.S3Service]: Factory method 'getS3Service' threw exception; nested exception is com.amazonaws.services.securitytoken.model.ExpiredTokenException: Token expired: current date/time 1603087602 must be before the expiration date/time1602216834 (Service: AWSSecurityTokenService; Status Code: 400; Error Code: ExpiredTokenException; Request ID: 0dca2b2b-3162-4b91-9c31-20059a1d9c4e; Proxy: null)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:596)
	at org.springframework.beans.factory.annotation.InjectionMetadata.inject(InjectionMetadata.java:90)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.postProcessProperties(AutowiredAnnotationBeanPostProcessor.java:374)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.populateBean(AbstractAutowireCapableBeanFactory.java:1411)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:592)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:515)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:320)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:318)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:845)
	at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:877)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:549)
	at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:141)
	at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:744)
	at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:391)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:312)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1215)
	at org.springframework.boot.SpringApplication.run(SpringApplication.java:1204)
	at ai.verta.modeldb.App.initializeServicesBaseOnArtifactStoreType(App.java:672)
	at ai.verta.modeldb.App.initializeServicesBaseOnDataBase(App.java:403)
	at ai.verta.modeldb.App.main(App.java:286)
Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'getS3Service' defined in ai.verta.modeldb.App: Bean instantiation via factory method failed; nested exception is org.springframework.beans.BeanInstantiationException: Failed to instantiate [ai.verta.modeldb.artifactStore.storageservice.s3.S3Service]: Factory method 'getS3Service' threw exception; nested exception is com.amazonaws.services.securitytoken.model.ExpiredTokenException: Token expired: current date/time 1603087602 must be before the expiration date/time1602216834 (Service: AWSSecurityTokenService; Status Code: 400; Error Code: ExpiredTokenException; Request ID: 0dca2b2b-3162-4b91-9c31-20059a1d9c4e; Proxy: null)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:627)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiateUsingFactoryMethod(ConstructorResolver.java:456)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.instantiateUsingFactoryMethod(AbstractAutowireCapableBeanFactory.java:1321)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1160)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:555)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:515)
	at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:320)
	at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:222)
	at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:318)
	at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:199)
	at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:277)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1251)
	at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1171)
	at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor$AutowiredFieldElement.inject(AutowiredAnnotationBeanPostProcessor.java:593)
	... 21 more
Caused by: org.springframework.beans.BeanInstantiationException: Failed to instantiate [ai.verta.modeldb.artifactStore.storageservice.s3.S3Service]: Factory method 'getS3Service' threw exception; nested exception is com.amazonaws.services.securitytoken.model.ExpiredTokenException: Token expired: current date/time 1603087602 must be before the expiration date/time1602216834 (Service: AWSSecurityTokenService; Status Code: 400; Error Code: ExpiredTokenException; Request ID: 0dca2b2b-3162-4b91-9c31-20059a1d9c4e; Proxy: null)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:185)
	at org.springframework.beans.factory.support.ConstructorResolver.instantiate(ConstructorResolver.java:622)
	... 34 more
Caused by: com.amazonaws.services.securitytoken.model.ExpiredTokenException: Token expired: current date/time 1603087602 must be before the expiration date/time1602216834 (Service: AWSSecurityTokenService; Status Code: 400; Error Code: ExpiredTokenException; Request ID: 0dca2b2b-3162-4b91-9c31-20059a1d9c4e; Proxy: null)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1828)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleServiceErrorResponse(AmazonHttpClient.java:1412)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1374)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1145)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:802)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:770)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:744)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:704)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:686)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:550)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:530)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.doInvoke(AWSSecurityTokenServiceClient.java:1652)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.invoke(AWSSecurityTokenServiceClient.java:1619)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.invoke(AWSSecurityTokenServiceClient.java:1608)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.executeAssumeRoleWithWebIdentity(AWSSecurityTokenServiceClient.java:1002)
	at com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClient.assumeRoleWithWebIdentity(AWSSecurityTokenServiceClient.java:972)
	at ai.verta.modeldb.artifactStore.storageservice.s3.S3Client.initializetWithTemporaryCredentials(S3Client.java:139)
	at ai.verta.modeldb.artifactStore.storageservice.s3.S3Client.<init>(S3Client.java:66)
	at ai.verta.modeldb.artifactStore.storageservice.s3.S3Service.<init>(S3Service.java:55)
	at ai.verta.modeldb.App.getS3Service(App.java:197)
	at ai.verta.modeldb.App$$EnhancerBySpringCGLIB$$2254d7c1.CGLIB$getS3Service$33(<generated>)
	at ai.verta.modeldb.App$$EnhancerBySpringCGLIB$$2254d7c1$$FastClassBySpringCGLIB$$c99ce303.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invokeSuper(MethodProxy.java:244)
	at org.springframework.context.annotation.ConfigurationClassEnhancer$BeanMethodInterceptor.intercept(ConfigurationClassEnhancer.java:363)
	at ai.verta.modeldb.App$$EnhancerBySpringCGLIB$$2254d7c1.getS3Service(<generated>)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.springframework.beans.factory.support.SimpleInstantiationStrategy.instantiate(SimpleInstantiationStrategy.java:154)
	... 35 more
java.lang.IllegalStateException: org.springframework.boot.web.servlet.context.AnnotationConfigServletWebServerApplicationContext@788e3702 has not been refreshed yet
	at org.springframework.context.support.AbstractApplicationContext.assertBeanFactoryActive(AbstractApplicationContext.java:1092)
	at org.springframework.context.support.AbstractApplicationContext.getBeansOfType(AbstractApplicationContext.java:1225)
	at org.springframework.boot.SpringApplication.exit(SpringApplication.java:1251)
	at ai.verta.modeldb.App.initiateShutdown(App.java:104)
	at ai.verta.modeldb.App.main(App.java:332)
```
So should we handle it and throw a proper error like `com.amazonaws.services.securitytoken.model.ExpiredTokenException: Token expired: current date/time 1603087602 must be before the expiration date/time1602216834 (Service: AWSSecurityTokenService; Status Code: 400; Error Code: ExpiredTokenException; Request ID: 0dca2b2b-3162-4b91-9c31-20059a1d9c4e; Proxy: null)`

Reply:
`token` is read from a file on disk. It makes sense to crash the whole system if that token is invalid since it means we have something misconfigured. I'd rather fail noisily than silently.

Reply:
Ok, then the current situation is good to throw noisily.. :)